Library/Homebrew/utils/formatter.rb

```diff
@@ -0,0 +1,46 @@
+require "utils/tty"
+
+module Formatter
+  module_function
+
+  def arrow(string, color: nil)
+    prefix("==>", string, color)
+  end
+
+  def headline(string, color: nil)
+    arrow("#{Tty.bold}#{string}#{Tty.reset}", color: color)
+  end
+
+  def success(string, label: nil)
+    label(label, string, :green)
+  end
+
+  def warning(string, label: nil)
+    label(label, string, :yellow)
+  end
+
+  def error(string, label: nil)
```

Comment:
Sorry if I'm making you go back and forth a bit, but after giving this some more thought, I think the combined handling of label and message string just complicates things without providing sufficient benefit. I think it would be best to keep the concatenation of label and message string on the call site and let `Formatter.error` et al. just wrap the given string like other `Formatter` methods. I.e. it should be used as follows:

```ruby
Formatter.error("FAILED")
"#{Formatter.error("Error:")} #{message}"
"#{Formatter.error("READ THIS:")} #{Formatter.url("https://example.com/")}"
```